### PR TITLE
Automate dependency checks with dependabot 

### PR DESCRIPTION
**Pull request by bishal7679**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature
- **Why was this change needed?** (You can also link to an open issue here)
to automate the dependency checking it is needed
Fixes :- #1596

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 2
+updates:
+  - package-ecosystem: maven
```
Comment (davidsoderberg) [inline .github/dependabot.yml:3]: should this one be maven is not that for Java?

**Comment by p-fernandez** [comment]
I have hard feelings about adding Dependabot integrated in the CI. Tends to be noisy and might ask to update dependencies that can cause breaking changes if there is an error in the version updated (e.g. recently cases of libraries only supporting ESM). Thus could become a blocker for the development, maintenance and contributions of Novu. Thoughts?
@davidsoderberg @scopsy @ainouzgali @djabarovgeorge

**Comment by scopsy** [comment]
@p-fernandez I aggree with you 100%. 

Dependapot was actually enabled a few months back, but it was creating a lot of PR noise and some strange errors. This is why I removed it. So also in this case, I wouldn't add dependabot for the library for those reasons. 

@bishal7679 thank you so much for your contribution, but from our previous experience we would prefer not to merge this at this point due to do the reason above 🙏

**Comment by p-fernandez** [comment]
Closing for the reasons given. Thanks for the effort @bishal7679 anyway. 🙌🏻

**Pull request closed without merging**